third_party/nvfuser/csrc/ir_internal_nodes.h

```diff
@@ -2181,4 +2229,121 @@ class TORCH_CUDA_CU_API NamedScalar : public Val {
   std::string name_;
 };
 
+class TORCH_CUDA_CU_API PadOp : public Expr {
+ public:
+  using Expr::Expr;
+
+  //! Pad a tensor as specified by a vector of integer scalars. For
+  //! the actual semantics, see the torch.pad documentation
+  PadOp(
+      IrBuilderPasskey passkey,
+      TensorView* out,
+      TensorView* inp,
+      const std::vector<Val*>& pad_widths);
+
+  NVFUSER_DECLARE_CLONE_AND_CREATE
+
+  virtual const char* getOpString() const override {
+    return "PadOp";
+  }
+
+  std::string toString(int indent_size = 0) const override;
+  std::string toInlineString(int indent_size = 0) const override;
+
+  Val* out() const {
+    return output(0);
+  }
+
+  Val* in() const {
+    return input(0);
+  }
+
+  std::vector<int> getPaddedAxes() const;
+
+  std::vector<Val*> getPadWidths() const;
+
+  std::pair<Val*, Val*> getPadWidths(int axis) const;
+};
+
+// Similar to at::indexing::Slice
+struct Slice {
+  Val* start = nullptr;
+  Val* stop = nullptr;
+  Val* step = nullptr;
+};
+
+class TORCH_CUDA_CU_API SliceOp : public Expr {
```

Comment:
Should slice also be an IterDomain expr?

Reply:
I don't think we need a different IterDomain expr as it can be expressed with `Resize`. 

However, there could be some situations where we may want to differentiate or keep information about how an IterDomain is resized. For example, when `Resize` is used for slicing, it is guaranteed that the output IterDomain has a smaller extent than the input, whereas it's the opposite when used for padding. Such information may be useful in some cases.